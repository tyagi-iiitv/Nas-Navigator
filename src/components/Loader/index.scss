.custom-loader {
    position: absolute;
    padding: 5px;
    z-index: 10;
    color: white;
    display: inline-block;
    right: 5%;
}