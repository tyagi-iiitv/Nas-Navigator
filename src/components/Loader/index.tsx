import React from "react";
import { Loader } from "semantic-ui-react";

import "./index.scss"

interface CustomLoaderProps {
    label: string;
    isActive: boolean;
    size: "big" | "small" | "mini" | "tiny" | "medium" | "large" | "huge" | "massive" | undefined;
}

export const CustomLoader: React.FC<CustomLoaderProps> = (props) => {
    if (props.isActive) {
        return <div className="custom-loader">
              
            <Loader content={ props.label }  size={props.size} inline active={props.isActive} />
        </div>
    }
    return <></>
}