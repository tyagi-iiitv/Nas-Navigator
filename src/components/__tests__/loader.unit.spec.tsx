import React from "react";
import { render } from "@testing-library/react";

import {CustomLoader as Loader} from "../Loader"

test('renders loader with correct label', () => {
    //Arrange
    const fakeProps = {label: 'my correct label', isActive: true, size: undefined};
    //Act 
    const {container, getByText} = render(<Loader {...fakeProps} />);
    const labelNode = getByText(fakeProps.label);
    //Assert
    expect(labelNode).toBeTruthy();
    expect(container).toMatchSnapshot();
})