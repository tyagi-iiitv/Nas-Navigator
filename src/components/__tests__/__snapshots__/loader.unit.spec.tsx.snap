// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loader with correct label 1`] = `
<div>
  <div
    class="custom-loader"
  >
    <div
      class="ui active text inline loader"
    >
      my correct label
    </div>
  </div>
</div>
`;
